### PR TITLE
refactor: split group_column.rs into multiple submodules

### LINKED ISSUE #13262: Reorganize the GroupColumn implementations into more coherent modules
### Is your feature request related to a problem or challenge?

While reviewing https://github.com/apache/datafusion/pull/12996 I noticed that all the implementations of `GroupColumn` for different types were in [`datafusion/physical-plan/src/aggregates/group_values/group_column.rs`](https://github.com/apache/datafusion/blob/main/datafusion/physical-plan/src/aggregates/group_values/group_column.rs) which is:
1. a single large file
2. Not obvious that it is like an implementation  detail of `GroupValuesColumn`: https://github.com/apache/datafusion/blob/main/datafusion/physical-plan/src/aggregates/group_values/column.rs#L94-L93

### Describe the solution you'd like

I would like to propose rearrainging the code: 
1. Rename https://github.com/apache/datafusion/blob/main/datafusion/physical-plan/src/aggregates/group_values/column.rs to https://github.com/apache/datafusion/blob/main/datafusion/physical-plan/src/aggregates/group_values/multi_column/mod.rs
2. Move the implementations in datafusion/physical-plan/src/aggregates/group_values/group_column.rs to different modules within https://github.com/apache/datafusion/blob/main/datafusion/physical-plan/src/aggregates/group_values/multi_column (like `primitive.rs`, `bytes.rs`, etc)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Chen-Yuan-Lai**
## Which issue does this PR close?

Closes #13262.

## What changes are included in this PR?

1. separate `/group_column.rs` into `/multi_column/primitive.rs`, `/multi_column/byte.rs`, and `/multi_column/byte_view.rs`
2. create `/multi_column/mod.rs` and place shared trait and function in this file

### CODE REVIEW

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jayzhan211** [comment]
@Chen-Yuan-Lai You can run `./dev/rust_lint.sh` to before push the code

File: datafusion/physical-plan/src/aggregates/group_values/column.rs [hunk 1]
```diff
@@ -17,10 +17,11 @@
 
 use std::mem::{self, size_of};
 
-use crate::aggregates::group_values::group_column::{
-    ByteGroupValueBuilder, ByteViewGroupValueBuilder, GroupColumn,
-    PrimitiveGroupValueBuilder,
-};
+use crate::aggregates::group_values::multi_column::byte::ByteGroupValueBuilder;
```
Comment (jayzhan211) [inline datafusion/physical-plan/src/aggregates/group_values/column.rs:20]: We prefer to merge them together

**Comment by Rachelint** [comment]
> @jayzhan211 thanks for your review! but the other PR (#13352) with the same issue has already been merged. @Rachelint you may need to reorganize codes again based on his change?

Yes.

**Comment by alamb** [comment]
https://github.com/apache/datafusion/pull/13392 -- I think has the follow on improvements

**Pull request closed without merging**